synapseclient/models/schema_organization.py

```diff
@@ -0,0 +1,731 @@
+    create_organization,
+    delete_json_schema,
+    delete_organization,
+    get_json_schema_body,
+    get_organization,
+    get_organization_acl,
+    list_json_schema_versions,
+    list_json_schemas,
+    list_organizations_sync,
+    update_organization_acl,
+)
+from synapseclient.core.async_utils import async_to_sync
+from synapseclient.models.mixins.json_schema import JSONSchemaVersionInfo
+from synapseclient.models.protocols.schema_organization_protocol import (
+    JSONSchemaProtocol,
+    SchemaOrganizationProtocol,
+)
+
+if TYPE_CHECKING:
+    from synapseclient import Synapse
+
+
+@dataclass()
+@async_to_sync
+class SchemaOrganization(SchemaOrganizationProtocol):
+    """
+    Represents an [Organization](https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/schema/Organization.html).
+
+    Attributes:
+        id: The ID of the organization
+        name: The name of the organization
+        created_on: The date this organization was created
+        created_by: The ID of the user that created this organization
+
+    Example:
+        from synapseclient.models import SchemaOrganization
+        from synapseclient import Synapse
+
+        syn = Synapse()
+        syn.login()
+
+        # Create a new org
+        org = SchemaOrganization("my.new.org.name")
+        org.create()
+
+        # Get the metadata and JSON Schemas for an existing org
+        org = SchemaOrganization("my.org.name")
+        org.get_async()
+        print(org)
+        schemas = org.get_json_schema_list()
+        print(schemas)
+    """
+
+    name: str
+    """The name of the organization"""
+
+    id: Optional[str] = None
+    """The ID of the organization"""
+
+    created_on: Optional[str] = None
+    """The date this organization was created"""
+
+    created_by: Optional[str] = None
+    """The ID of the user that created this organization"""
+
+    def __post_init__(self) -> None:
+        self._check_name(self.name)
+
+    def __repr__(self):
+        return f"SchemaOrganization(name={self.name!r})"
+
+    async def get_async(self, synapse_client: Optional["Synapse"] = None) -> None:
+        """
+        Gets the metadata from Synapse for this organization
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor
+
+        Example:
+            from synapseclient.models import SchemaOrganization
+            from synapseclient import Synapse
+            import asyncio
+
+            syn = Synapse()
+            syn.login()
+
+            org = SchemaOrganization("my.org.name")
+            asyncio.run(org.get_async())
+
+        """
+        if self.id:
+            return
+        response = await get_organization(self.name, synapse_client=synapse_client)
+        self._fill_from_dict(response)
+
+    # Should this be named 'store_async'?
```

Comment:
Yes - For consistency I'd prefer that things be `store` rather than `create`, unless it really does need 2 separate methods `update` and `create`, but in this case it doesn't look like it does need to be separate